kv/dist_sender.go

```diff
@@ -353,7 +344,8 @@ func (ds *DistSender) sendRPC(
 	ba roachpb.BatchRequest,
 ) (*roachpb.BatchResponse, error) {
 	if len(replicas) == 0 {
-		return nil, noNodeAddrsAvailError{}
+		return nil, roachpb.NewSendError(
```

Comment:
This change allows the system to properly retry such errors instead of failing the request to the distributed sender.